src/dftbp/mixer/andersonmixer.F90

```diff
@@ -73,61 +103,51 @@ module dftbp_mixer_andersonmixer
 
     !> Stored prev. charge differences
     ${TYPE}$(dp), allocatable :: prevQDiff(:,:)
+      contains
+        procedure :: reset => TAndersonMixer${LABEL}$_reset
+        procedure :: mix1D => TAndersonMixer${LABEL}$_mix
   end type TAndersonMixer${LABEL}$
-#:endfor
 
+#:endfor
 
 contains
 
 #:for NAME, TYPE, LABEL in FLAVOURS
-  !> Creates an Andersom mixer instance.
-  subroutine TAndersonMixer${LABEL}$_init(this, nGeneration, mixParam, initMixParam, convMixParam,&
-      & omega0)
-
-    !> Initialized Anderson mixer on exit
-    type(TAndersonMixer${LABEL}$), intent(out) :: this
-
-    !> Nr. of generations (including actual) to consider
-    integer, intent(in) :: nGeneration
 
-    !> Mixing parameter for the general case
-    real(dp), intent(in) :: mixParam
+  !> Creates an Anderson mixer instance.
+  subroutine TAndersonMixer${LABEL}$_init(this, mixerInp)
 
-    !> Mixing parameter for the first nGeneration-1 cycles
-    real(dp), intent(in) :: initMixParam
-
-    !! Convergence dependent mixing parameters. Given as 2 by n array of tolerance and mixing
-    !! factors pairs. The tolerances (Euclidean norm of the charge diff. vector) must follow each
-
-    !> Other in decreasing order.
-    !! Mixing parameters given here eventually override mixParam or initMixParam.
-    real(dp), intent(in), optional :: convMixParam(:,:)
+    !> Initialized Anderson mixer on exit
+    type(TAndersonMixer${LABEL}$), allocatable, intent(out) :: this
```

Comment:
init routines should not allocate the instance (otherwise we would be never able to use it with static instances).
```suggestion
    type(TAndersonMixer${LABEL}$), intent(out) :: this
```